xkafka/consumer_simple.go

```diff
@@ -0,0 +1,112 @@
+package xkafka
+
+import (
+	"context"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+)
+
+// SimpleConsumer is a simple consumer that consumes messages from a Kafka topic.
+type SimpleConsumer struct {
+	config       ConsumerConfig
+	kafka        *kafka.Consumer
+	middlewares  []middleware
+	errorHandler ErrorHandler
+}
+
+// NewSimpleConsumer creates a new SimpleConsumer instance.
+func NewSimpleConsumer(cfg ConsumerConfig) (*SimpleConsumer, error) {
+	cfg.SetDefaults()
+
+	kafkaCfg := &kafka.ConfigMap{
+		"bootstrap.servers":           cfg.Brokers,
+		"group.id":                    cfg.Group,
+		"enable.auto.commit":          cfg.AutoCommit,
+		"auto.offset.reset":           cfg.TopicOffset,
+		"socket.keepalive.enable":     true,
+		"session.timeout.ms":          int(cfg.SessionTimeout.Milliseconds()),
+		"heartbeat.interval.ms":       int(cfg.HeartbeatInterval.Milliseconds()),
+		"auto.commit.interval.ms":     int(cfg.AutoCommitInterval.Milliseconds()),
+		"metadata.request.timeout.ms": cfg.MetadataRequestTimeout,
+		"socket.timeout.ms":           int(cfg.SocketTimeout.Milliseconds()),
+	}
+
+	consumer, err := kafka.NewConsumer(kafkaCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	err = consumer.SubscribeTopics(cfg.Topics, nil)
```

Comment:
Will this communicate with kafka broker? If yes, any communication to broker should happen after the `Start`/`Run` method is called. It gives more flexibility to user on when to start the communication in their flows.